Upstream-Status: Pending

Description: Turn #error into exceptions
 So the library can be built on non-Linux platforms too, although can't
 guarauntee its functionality regarding that piece of code.
Forwarded: not-needed
--- a/system/core/base/file.cpp
+++ b/system/core/base/file.cpp
@@ -422,7 +422,8 @@
   path[PATH_MAX - 1] = 0;
   return path;
 #else
-#error unknown OS
+#include <stdexcept>
+  throw std::runtime_error(std::string("Unknown OS!"));
 #endif
 }
 
